openff/interchange/_tests/unit_tests/operations/test_combine.py

```diff
@@ -219,3 +219,21 @@ def test_constraint_key_collision(self, parsley, sage, ethanol, flip_order, hand
             numpy.vstack(arrays_before),
             array_after_combine,
         )
+
+    @pytest.mark.parametrize("flip_order", [False, True])
+    def test_charge_cache_behavior(self, sage, ethanol, caffeine, flip_order):
+        int1 = sage.create_interchange(ethanol.to_topology())
+        int2 = sage.create_interchange(caffeine.to_topology())
+
+        for interchange_ in (int1, int2):
+            assert len(interchange_["Electrostatics"]._get_charges()) == interchange_.topology.n_atoms
+
+        if flip_order:
+            int3 = int2.combine(int1)
+        else:
+            int3 = int1.combine(int2)
+
+        assert int3["Electrostatics"]._charges_cached is False
+
+        for interchange_ in (int1, int2):
+            assert len(interchange_["Electrostatics"]._get_charges()) == interchange_.topology.n_atoms
```

Comment:
NON-BLOCKING - Feel free not to include this, but if I'm understanding correctly I think a short comment to clarify why `int3` is not included in this check could be helpful for future readers. 

```suggestion 
        # Check that the input caches are intact
        for interchange_ in (int1, int2):
            assert len(interchange_["Electrostatics"]._get_charges()) == interchange_.topology.n_atoms
```